llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -2379,6 +2381,11 @@ bool VPlanTransforms::tryAddExplicitVectorLength(
       CanonicalIVIncrement->getDebugLoc(), "index.evl.next");
   EVLPhi->addOperand(NextEVLIV);
 
+  VPValue *NextAVL = Builder.createOverflowingOp(
+      Instruction::Sub, {AVLPhi, OpVPEVL}, {/*hasNUW=*/true, /*hasNSW=*/false},
```

Comment:
Can you explain why it has nuw but not nsw?

Reply:
AFAIK the trip count is always an unsigned integer, and we're decrementing it downwards towards zero by EVL each time.

EVL should never be larger than the original trip count so it should never have unsigned wrap.

I didn't add nsw because I think we can go from SINT_MIN to e.g. SINT_MIN-1 which wraps